src/Component.php

```diff
@@ -20,30 +20,39 @@ public function run(): void
             $client = new Client(['token' => $config->getToken(), 'url' => $config->getUrl()]);
                 $tokenInfo = $client->verifyToken();
             if (count($tokenInfo['bucketPermissions']) > 1) {
-                throw new UserException("The token has too broad permissions.");
+                throw new UserException('The token has too broad permissions.');
             }
             $bucket = array_keys($tokenInfo['bucketPermissions'])[0];
             if ($tokenInfo['bucketPermissions'][$bucket] !== 'write') {
-                throw new UserException("The token does not have write permissions to the bucket " . $bucket);
+                throw new UserException('The token does not have write permissions to the bucket ' . $bucket);
             }
             foreach ($config->getInputTables() as $table) {
-                $this->getLogger()->info("Processing table " . $table['destination']);
+                $this->getLogger()->info('Processing table ' . $table['destination']);
                 $manifest = $this->getManifestManager()->getTableManifest($table['destination']);
+                $primaryKey = $manifest['primary_key'] ?? [];
                 $csv = new CsvFile($this->getDataDir() . '/in/tables/' . $table['destination']);
                 $tableId = $bucket . '.' . $table['destination'];
                 if ($client->tableExists($tableId)) {
-                    $client->writeTableAsync($tableId, $csv);
+                    $tableInfo = $client->getTable($tableId);
+                    if ($tableInfo['primaryKey'] != $primaryKey) {
+                        throw new UserException(
+                            'Incompatible primary key encountered, target table has: ' .
```

Comment:
Co nějakou hlášku podobnou tomuhle? Ať to není pokaždý jinak.

https://github.com/keboola/output-mapping/blob/15c227217e0e9eebf875620c65a7b4f2a580d650/Writer/Writer.php#L652

Reply:
output mapping bych tady nezminoval, protoze v UI neni, ale jmeno tabulky tam pridam!

Reply:
určitě! jen spíš ať to drží jednotnej jazyk, když je to stejná chyba.